### PR TITLE
Fetch datacenter resource by id

### PR DESCRIPTION
**Pull request by Ntr0**
The datacenter resource does accept the datacenter id as parameter. This will take precendence over location and name, failing if datacenter with given id does not match name and location (if present). The behaviour for using name and location only does not change.

### CODE REVIEW

**Comment by jasmingacic** [comment]
@Ntr0 This is looking good. Before we merge it can you please paste output of `make testacc` 
Thank you

File: profitbricks/config.go [hunk 1]
```diff
@@ -10,11 +10,17 @@ type Config struct {
 	Password string
 	Endpoint string
 	Retries  int
+	Token    string
 }
 
 // Client() returns a new client for accessing ProfitBricks.
 func (c *Config) Client() (*profitbricks.Client, error) {
-	client := profitbricks.NewClient(c.Username, c.Password)
+	var client *profitbricks.Client
```
Comment (jasmingacic) [inline profitbricks/config.go]: This section is already covered with #44 so can you please remove it.

File: profitbricks/provider.go [hunk 2]
```diff
@@ -15,16 +15,25 @@ func Provider() terraform.ResourceProvider {
 	return &schema.Provider{
 		Schema: map[string]*schema.Schema{
 			"username": {
-				Type:        schema.TypeString,
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("PROFITBRICKS_USERNAME", nil),
-				Description: "ProfitBricks username for API operations.",
+				Type:          schema.TypeString,
```
Comment (jasmingacic) [inline profitbricks/provider.go]: This section is already covered with #44 so can you please remove it.

**Comment by Ntr0** [comment]
Hi @jasmingacic 
I removed the commits from #44 
Thanks for reviewing

**Comment by jasmingacic** [comment]
`https://github.com/terraform-providers/terraform-provider-profitbricks/releases/tag/v1.4.1`

**Pull request merged**